### PR TITLE
:book:  Remove "latest release"

### PR DESCRIPTION
**Pull request by shivam-51**
**What this PR does / why we need it**:
The term "latest release" does not seem relevant anymore and is redundant as the latest release now is v0.4.1.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/user/quick-start.md [hunk 1]
```diff
@@ -96,7 +96,7 @@ The clusterctl CLI tool handles the lifecycle of a Cluster API management cluste
 {{#tab linux}}
 
 #### Install clusterctl binary with curl on linux
-Download the latest release; for example, to download version v0.3.0 on linux, type:
+To download version v0.3.0 on linux, type:
```
Comment (fabriziopandini) [inline docs/book/src/user/quick-start.md:99]: The curl instruction we are not downloading v0.3.0, but the latest release in the 0.3 series, so IMO we should drop the example, not the first part of the sentence.

I suggest to align this to what we have on https://cluster-api.sigs.k8s.io/user/quick-start.html (
`Download the latest release; on linux, type:`)

Same below

Comment (vincepri) [reply docs/book/src/user/quick-start.md:99]: +1 to keep this text generic

Comment (cpanato) [reply docs/book/src/user/quick-start.md:99]: +1

Comment (shivam-51) [reply docs/book/src/user/quick-start.md]: On a second thought, this makes sense. Updated.

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**